transformer/app.js

```diff
@@ -101,21 +118,122 @@ function transformIssue(body, type, accountId) {
+  const results = await Promise.allSettled(promises);
+  return results.filter((result) => (result.status === 'fulfilled' && result.value.Count > 0)).map((result) => {
+    console.debug(JSON.stringify(result.value.Items[0].ID.S));
+    return result.value.Items[0].ID.S;
+  });
+};
+
+const sendUpdateCommand = async (updateBody) => {
+  const promises = updateBody.map((body) => hubClient.send(new BatchUpdateFindingsCommand(body)));
+  const results = await Promise.allSettled(promises);
+  console.debug(`Unfiltered results (promises): ${JSON.stringify(results)}`);
+  const resultsFulfilled = results.filter((result) => (result.status === 'fulfilled')).map((result) => result.value.ProcessedFindings);
+  const resultsFailed = results.filter((result) => (result.status !== 'fulfilled')).map((result) => result.value.ProcessedFindings);
+  console.debug(`resultsFulfilled: ${JSON.stringify(resultsFulfilled)}`);
+  return {
+    SuccessCount: resultsFulfilled.length,
+    SuccessfulFindings: resultsFulfilled,
+    FailedFindings: resultsFailed.length,
+    FailedCount: resultsFailed,
+  };
+};
+
 export async function lambdaHandler(event, context) {
   let response;
+  console.debug(`Lambda triggered by SQS message: ${JSON.stringify(event)}`);
   try {
     const accountId = process.env.USE_DEV_ACCOUNT_ID === 'true' ? process.env.DEV_ACCOUNT_ID : context.invokedFunctionArn.split(':')[4];
-    const parsedBody = JSON.parse(event.Records[0].body);
-    let findings = [];
-    const type = parsedBody.type.toLowerCase();
-    findings = transformIssue(parsedBody, type, accountId);
-    console.log(`Findings to send to Hub: ${JSON.stringify(findings)}`);
-    const hubResponse = await hubClient.send(new BatchImportFindingsCommand({ Findings: findings }));
-    console.log(`Security Hub response: ${JSON.stringify(hubResponse)}`);
+    const findingsCollection = event.Records.map((sqsEvent) => {
+      const parsedBody = JSON.parse(sqsEvent.body);
+      return transformIssue(parsedBody, accountId);
+    });
+    console.debug(`Complete findings list: ${JSON.stringify(findingsCollection.flat())}`);
+    const completeIdsCollection = findingsCollection.flat().map((finding) => finding.Id);
+    console.debug(`Unfiltered finding IDs collection: ${JSON.stringify(completeIdsCollection)}`);
+    const existingFindingIDsInDB = await verifyFindingIds(completeIdsCollection);
+    console.debug(`Existing finding IDs in the DB: ${JSON.stringify(existingFindingIDsInDB)}`);
+    const newFindingIDs = completeIdsCollection.filter((id) => !existingFindingIDsInDB.includes(id));
+    console.debug(`New findings to import (IDs only): ${JSON.stringify(newFindingIDs)}`);
+    const existingFindingsToUpdate = findingsCollection.flat().filter((item) => existingFindingIDsInDB.includes(item.Id));
+    const newFindingsToImport = findingsCollection.flat().filter((item) => !existingFindingIDsInDB.includes(item.Id));
```

Comment:
Please add blank lines between these groups of code to make them easier to read.

Reply:
This group of code is good candidate to be extracted into a function. Between L205 to L213, we really only care about the 2 list of findings. The code in between are details that can be hidden inside a function.

Reply:
I think it's better to leave them as is, just add spaces for better readability.